front/components/app/blocks/Input.tsx

```diff
@@ -63,21 +123,46 @@ export default function Input({
       onBlockNew={onBlockNew}
       canUseCache={false}
     >
-      <div className="mx-4 flex flex-col sm:flex-row sm:space-x-2">
-        <div className="flex flex-row items-center space-x-2 text-sm font-medium leading-8 text-gray-700">
+      <div>
+        <div>
           {!((!block.config || !block.config.dataset) && readOnly) ? (
-            <>
-              <div className="flex flex-initial">dataset:</div>
+            <div className="flex flex-row items-center space-x-2 text-sm font-medium leading-8 text-gray-700">
+              Dataset:&nbsp;
               <DatasetPicker
                 owner={owner}
                 app={app}
                 dataset={block.config ? block.config.dataset : ""}
                 onDatasetUpdate={handleSetDataset}
                 readOnly={readOnly}
               />
-            </>
+              {block.config && block.config.dataset ? (
+                <Button
+                  variant="secondary"
+                  onClick={() => {
+                    window.location.href = `/w/${owner.sId}/a/${app.sId}/datasets/${block.config.dataset}`;
+                  }}
+                  icon={PencilSquareIcon}
+                  label="Edit schema"
+                  size="xs"
+                />
+              ) : null}
+            </div>
           ) : null}
         </div>
+
+        {datasetWithData && datasetWithData.schema ? (
+          <div className="max-h-[800px] overflow-y-auto">
```

Comment:
I'm not a massive fan of the double scrolling here?

Reply:
I'm still not 100% convinced by this :) Let's discuss more :+1:

Reply:
Last try with one quick change -> modal on the side. 2 clicks to go to schema edition but this fixes the 'too long' issue and allows iterating. Wdyt @spolu ?

https://github.com/dust-tt/dust/assets/1189312/e892697c-3d8a-44a8-b0db-7ecbdf5d90e2